src/mpyl/stages/discovery.py

```diff
@@ -71,7 +72,7 @@ def _are_invalidated(
 
 
 def find_invalidated_projects_for_stage(
-    all_projects: set[Project], stage: Stage, change_history: list[Revision]
+    all_projects: set[Project], stage: str, change_history: list[Revision]
```

Comment:
Yeah adding an enum as noted above would definitely be an improvement imo, because `Stage` being a class in some places/functions and a `str` in others can be quite confusing. Especially when used almost next to eachother.

Reply:
https://github.com/Vandebron/mpyl/pull/255#discussion_r1359759603